python/lsst/pex/config/config.py

```diff
@@ -729,13 +729,14 @@ def __get__(self, instance, owner=None, at=None, label="default"):
             # try statements are almost free in python if they succeed
             try:
                 return instance._storage[self.name]
-            except AttributeError:
+            except AttributeError as e:
                 if not isinstance(instance, Config):
                     return self
                 else:
-                    raise AttributeError(
+                    e.add_note(
```

Comment:
I thought some of the intent here was to hide the implementation details that no one should need to know.  Adding a note and reraising exposes those in the traceback.

Reply:
Should use `raise from None` then.